### PR TITLE
Closes #5682 RocketCDN CTA banner displayed in cdn tab on one dot com hosting

### LINKED ISSUE #5682: ONE - RocketCDN's CTA banner is displayed in the CDN tab
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version :heavy_check_mark: `3.12.5`
- Used the search feature to ensure that the bug hasn’t been reported before :heavy_check_mark: 

**Describe the bug**

In `3.12.5` and for customers on ONE's network we still display RocketCDN's banner:
image

**To Reproduce**

1. Install WP Rocket `3.12.5` on our ONE test site.
2. Go to the CDN tab.
3. See the banner.

**Expected behavior**

RocketCDN's banner should not be displayed; the same goes for its subscription status in WP Rocket's dashboard.

**Backlog Grooming (for WP Media dev team use only)**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description
This PR removes the RocketCDN CTA banner when ONE.com cdn is enabled.

Fixes #5682 

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

NO

## How Has This Been Tested?

Automated Tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by remyperona — APPROVED** [review]

**Comment by vmanthos** [comment]
@jeawhanlee The CTA in the CDN tab is hidden. :+1: 

There are two more places we need to hide RocketCDN info:
1. 
image

coming from:
https://github.com/wp-media/wp-rocket/blob/55ac46292a05931c202d31881f7fdcf5467a5200/inc/Engine/CDN/RocketCDN/views/promote-notice.php#L10-L14

and the following in WP Rocket's dashboard:
2. 
image

coming from:
https://github.com/wp-media/wp-rocket/blob/55ac46292a05931c202d31881f7fdcf5467a5200/inc/Engine/CDN/RocketCDN/views/dashboard-status.php#L18-L37

Can you please look into this? :pray:

File: inc/ThirdParty/Hostings/OneCom.php [hunk 1]
```diff
@@ -162,4 +165,31 @@ public function build_cname(): string {
...
+	}
+
+	/**
+	 * Remove RocketCDN promotion notice.
+	 *
+	 * @return boolean
+	 */
+	public function maybe_remove_rocketcdn_promotion_notice(): bool {
+		return ! $this->is_oc_cdn_enabled();
+	}
```
Comment (remyperona) [inline inc/ThirdParty/Hostings/OneCom.php]: I think we could re-use one method as a callback for the 3 filters instead

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Review by vmanthos — COMMENTED** [review]
With the recent [change](https://github.com/wp-media/wp-rocket/pull/5712) in the way we detect if  ONE's CDN is enabled, all RocketCDN CTA/info will be displayed if
* the Performance Cache is deactivated `OR`
* it's activated but the CDN option is disabled

In these scenarios, ONE customers will be able to purchase a RocketCDN subscription.

@piotrbak I'm guessing this isn't what we want. Can you please confirm?

@jeawhanlee Depending on Piotr's reply you may need to create a new method to detect if the site is hosted on ONE and hook that to the UI hooks you are using to hide RocketCDN CTA/info.

Also, some tests are[ failing](https://github.com/wp-media/wp-rocket/actions/runs/4012993264). Can you please look into this? :pray:

**Review by remyperona — APPROVED** [review]

**Review by vmanthos — APPROVED** [review]
Working as expected. :+1:

**Pull request merged**